### PR TITLE
2016-day-10

### PR DESCRIPTION
**Pull request by Herrera93**
- Created library to manage bots
- To solve day 10

## PR - Merge Checklist
- [ ] CR'd
- [ ] Workflow: Trello card is in "Code Review" column
- [ ] README updated or N/A
- [ ] Benchmarks run and compared with master or N/A

### CODE REVIEW

**Comment by Herrera93** [comment]
[2016-day-10-part-1](https://trello.com/c/xJaCA3UA/115-2016-day-10-part-1)

**Comment by Herrera93** [comment]
[2016-day-10-part-2](https://trello.com/c/KX5eokUl/140-2016-day-10-part-2)

File: src/year2016/day_10.rs [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+#[derive(Copy, Clone, Debug)]
+enum Instruction {
+    LowToOutput(i32),
+    LowToBot(i32),
+    HighToOutput(i32),
+    HighToBot(i32),
+}
+
+impl Instruction {
+    fn from_instruction(value_type: &str, receiver: &str, receiver_id: i32) -> Self {
```
[the remark this answers is not in the document]
Comment (Herrera93) [reply src/year2016/day_10.rs]: Yes, you're right. Let me change that.

**Comment by Herrera93** [comment]
@lifehug Yes, a more idiomatic way to handle `unwrap()` is to use the `?`, in this case I used it when parsing `string` to `i32`, but didn't use it when getting the regex capture since it would require to define a custom error like we did day 21 2015 and I thought that it was going to be a lot of code but I could do it if it's better that way.

**Comment by lifehug** [comment]
I don't think it's important to make that change now — more personally curious. You're teaching me 👍

**Pull request merged**